Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -647,546 +552,351 @@ protected virtual void Dispose(bool disposing)
-
-                                    // Store interface name so we can use the name in Collections.
-                                    _interfaceNames[adapter.Description.ToLower(CultureInfo.InvariantCulture)] = tag;
-                                    _interfaceNames["eth" + tag.ToString(CultureInfo.InvariantCulture)] = tag;
-                                }
-                            }
-                        }
-#pragma warning disable CA1031 // Do not catch general exception types
-                        catch (Exception ex)
-                        {
-                            // Ignore error, and attempt to continue.
-                            _logger.LogError(ex, "Error encountered parsing interfaces.");
-                        }
-#pragma warning restore CA1031 // Do not catch general exception types
-                    }
-                }
-                catch (Exception ex)
-                {
-                    _logger.LogError(ex, "Error in InitialiseInterfaces.");
-                }
+        /// <inheritdoc/>
+        public bool IsInLocalNetwork(IPAddress address)
+        {
+            ArgumentNullException.ThrowIfNull(address);
 
-                // If for some reason we don't have an interface info, resolve our DNS name.
-                if (_interfaceAddresses.Count == 0)
-                {
-                    _logger.LogError("No interfaces information available. Resolving DNS name.");
-                    IPHost host = new IPHost(Dns.GetHostName());
-                    foreach (var a in host.GetAddresses())
-                    {
-                        _interfaceAddresses.AddItem(a);
-                    }
+            // See conversation at https://github.com/jellyfin/jellyfin/pull/3515.
+            if (TrustAllIpv6Interfaces && address.AddressFamily == AddressFamily.InterNetworkV6)
+            {
+                return true;
+            }
 
-                    if (_interfaceAddresses.Count == 0)
-                    {
-                        _logger.LogWarning("No interfaces information available. Using loopback.");
-                    }
-                }
+            // As private addresses can be redefined by Configuration.LocalNetworkAddresses
+            var match = CheckIfLanAndNotExcluded(address);
 
-                if (IsIP4Enabled)
-                {
-                    _interfaceAddresses.AddItem(IPNetAddress.IP4Loopback);
-                }
+            return address.Equals(IPAddress.Loopback) || address.Equals(IPAddress.IPv6Loopback) || match;
+        }
 
-                if (IsIP6Enabled)
-                {
-                    _interfaceAddresses.AddItem(IPNetAddress.IP6Loopback);
-                }
+        private bool CheckIfLanAndNotExcluded(IPAddress address)
+        {
+            bool match = false;
+            foreach (var lanSubnet in _lanSubnets)
+            {
+                match |= lanSubnet.Contains(address);
```

Comment:
Please don't use |= and &=

Reply:
Why should it not be used and what would be a better alternative?